kafka-client/src/commonMain/kotlin/com/icemachined/kafka/common/serialization/JsonDeserializer.kt

```diff
@@ -0,0 +1,29 @@
+package com.icemachined.kafka.common.serialization
+
+import com.icemachined.kafka.clients.consumer.Headers
+import com.icemachined.kafka.common.header.KafkaHeaders
+
+import kotlin.reflect.KType
+import kotlinx.cinterop.toKString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+
+/**
+ * JsonDeserializer
+ */
+class JsonDeserializer<T>(
+    private val typesMap: Map<String, KType>
+) : Deserializer<T> {
+    override fun deserialize(
+        data: ByteArray,
+        topic: String?,
+        headers: Headers?
+    ): T? {
+        val typeId = headers?.lastHeader(KafkaHeaders.KTYPE_ID)?.value?.toKString()
+        return typeId?.let {
+            typesMap[typeId]?.let { targetType ->
```

Comment:
maybe print a warning at least if we received a message which doesn't supported by our listener ?

Reply:
good notice, now i throw DeserializationException